pkg/readiness/ready_tracker.go

```diff
@@ -800,15 +874,23 @@ func (t *Tracker) trackConfigAndSyncSets(ctx context.Context) error {
 
 		// Set expectations for individual cached resources
 		dt := t.ForData(gvkCpy)
-		go func() {
+		t.dataTrackers.Go(ctx, gvk.String(), func(ctx context.Context) error {
 			err := t.trackData(ctx, gvkCpy, dt)
 			if err != nil {
 				log.Error(err, "aborted trackData", "gvk", gvkCpy)
+				if *crashOnFailureFetchingExpectations {
+					return err
+				}
 			}
-		}()
+			return nil
+		})
 	}
 
-	return nil
+	if retErr == nil {
```

Comment:
why not `return retErr` here?

Reply:
had to do with some edge case with golang errors & nil comparisons I talked about with @maxsmythe but for this case I think we agreed it can just be `return retErr`